plugins/statistics.koplugin/main.lua

```diff
@@ -1862,38 +2102,75 @@ function ReaderStatistics:onPageUpdate(pageno)
     if self:isDocless() or not self.is_enabled then
         return
     end
-    self.curr_page = pageno
-    self.pages_stats[TimeVal:now().sec] = pageno
-    local mem_read_pages = 0
-    local mem_read_time = 0
-    if util.tableSize(self.pages_stats) > 1 then
-        local sorted_performance = {}
-        for time, page in pairs(self.pages_stats) do
-            table.insert(sorted_performance, time)
+
+    -- We only care about *actual* page turns ;)
+    if self.curr_page == pageno then
+        return
+    end
+
+    self.pageturn_count = self.pageturn_count + 1
+    local now_ts = TimeVal:now().sec
+
+    -- Get the previous page's last timestamp (if there is one)
+    local page_data = self.page_stat[self.curr_page]
+    -- This is a list of tuples, in insertion order, we want the last one
```

Comment:
(Haven't re-read this part. Can I assume it hasn't changed much since the other PR?)

Reply:
It hasn't, at all ;).